opm/polymer/fullyimplicit/SimulatorFullyImplicitBlackoilPolymer_impl.hpp

```diff
@@ -115,6 +121,134 @@ namespace Opm
                                             timer.simulationTimeElapsed() + timer.currentStepLength(),
                                             polymer_inflow_c);
         well_state.polymerInflow() = polymer_inflow_c;
+
+        computeRepRadiusPerfLength(BaseType::eclipse_state_, timer.currentStepNum(), BaseType::grid_, wells_rep_radius_, wells_perf_length_);
+    }
+
+
+    template <class GridT>
+    void SimulatorFullyImplicitBlackoilPolymer<GridT>::
+    setupCompressedToCartesian(const int* global_cell, int number_of_cells,
+                               std::map<int,int>& cartesian_to_compressed )
+    {
+        if (global_cell) {
+            for (int i = 0; i < number_of_cells; ++i) {
+                cartesian_to_compressed.insert(std::make_pair(global_cell[i], i));
+            }
+        }
+        else {
+            for (int i = 0; i < number_of_cells; ++i) {
+                cartesian_to_compressed.insert(std::make_pair(i, i));
+            }
+        }
+
+    }
+
+
+    template <class GridT>
+    void SimulatorFullyImplicitBlackoilPolymer<GridT>::
+    computeRepRadiusPerfLength(const Opm::EclipseStateConstPtr eclipseState,
+                               const size_t                    timeStep,
+                               const GridT&                    grid,
+                               std::vector<double>&            wells_rep_radius,
+                               std::vector<double>&            wells_perf_length)
+    {
+
+        int number_of_cells = Opm::UgGridHelpers::numCells(grid);
+        const int* global_cell = Opm::UgGridHelpers::globalCell(grid);
+        const int* cart_dims = Opm::UgGridHelpers::cartDims(grid);
+        auto cell_to_faces = Opm::UgGridHelpers::cell2Faces(grid);
+        auto begin_face_centroids = Opm::UgGridHelpers::beginFaceCentroids(grid);
+
+        if (eclipseState->getSchedule()->numWells() == 0) {
+            OPM_MESSAGE("No wells specified in Schedule section, "
+                        "initializing no wells");
+            return;
+        }
+
+        wells_rep_radius.clear();
+        wells_perf_length.clear();
+
```

Comment:
Basically, we are not sure what the size will be. I reserve based on the previous size. Please suggest if it is OK.